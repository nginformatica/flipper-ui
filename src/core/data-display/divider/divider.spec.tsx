import React from 'react'
import { render, screen } from '@testing-library/react'
import { Divider } from '.'

describe('Divider', () => {
    it('should render', () => {
        render(<Divider data-testid='divider' />)

        const divider = screen.getByTestId('divider')

        expect(divider).toBeDefined()
    })

    it('should render with custom style', () => {
        render(
            <Divider
                data-testid='divider'
                margin={10}
                padding={5}
                style={{ backgroundColor: 'blue' }}
            />
        )

        const divider = screen.getByTestId('divider')

        expect(divider).toHaveProperty('style.margin', '10px')
        expect(divider).toHaveProperty('style.padding', '5px')
        expect(divider).toHaveProperty('style.backgroundColor', 'blue')
    })

    it('should match snapshot', () => {
        const { container } = render(<Divider />)

        expect(container).toMatchSnapshot()
    })
})
