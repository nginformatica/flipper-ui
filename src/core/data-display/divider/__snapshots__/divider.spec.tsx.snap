// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should match snapshot 1`] = `
<div>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</div>
`;
